packages/react-components/react-table/src/hooks/useInteractiveKeyboardResizing.ts

```diff
@@ -0,0 +1,103 @@
+import { useEventCallback } from '@fluentui/react-utilities';
+import * as React from 'react';
+import { useKeyboardNavigationContext } from '../contexts/keyboardNavigationContext';
+import { ColumnResizeState, TableColumnId } from './types';
+
+export function useInteractiveKeyboardResizing(columnResizeState: ColumnResizeState) {
+  const DECREASE_WIDTH = 'ArrowLeft';
+  const INCREASE_WIDTH = 'ArrowRight';
+  const SPACEBAR = ' ';
+  const ENTER = 'Enter';
+  const ESC = 'Escape';
+
+  const STEP = 20;
+  const PRECISION_MODIFIER = 'Shift';
+  const PRECISION_FACTOR = 1 / 4;
+
+  const { setNavigationGroupParams, defaultNavigationGroupParams } = useKeyboardNavigationContext();
+  const columnId = React.useRef<TableColumnId>();
+
+  const columnResizeStateRef = React.useRef<ColumnResizeState>(columnResizeState);
```

Comment:
because I'm using `window.addEventListener` and the handler has the old state reference in closure, updating the state will not update the handler. Using a ref here works around this, because the ref.current will always be latest state